src/Microsoft.AspNet.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -174,10 +113,12 @@ public OAuthHandler(HttpClient backchannel)
             requestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
             requestMessage.Content = requestContent;
             var response = await Backchannel.SendAsync(requestMessage, Context.RequestAborted);
-            response.EnsureSuccessStatusCode();
-            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
-
-            return new OAuthTokenResponse(payload);
+            if (response.IsSuccessStatusCode)
```

Comment:
How about we add an error string field to the OAuthToken response class? Looks like today we just use null token response to signal error.